hyprtester/src/tests/main/window.cpp

```diff
@@ -131,6 +131,123 @@ static void testSwapWindow() {
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        expectActiveWindow("kitty_A", '2');
+
+        // Dispatch-focus a different window
+        OK(getFromSocket("/dispatch focuswindow class:kitty_B"));
+        expectActiveWindow("kitty_B", '0');
+
+        // Make a window that will request focus
+        if (!spawnKittyActivating())
+            return false;
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        OK(getFromSocket("/dispatch fullscreen 0"));
+        expectActiveWindow("kitty_A", '2');
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+        expectActiveWindow("kitty_activating", '0');
+    }
+
+    // Take over on conflict
+    {
+        OK(getFromSocket("/keyword misc:on_focus_under_fullscreen 1"));
+
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        OK(getFromSocket("/dispatch fullscreen 0"));
+        expectActiveWindow("kitty_A", '2');
+
+        // Dispatch-focus the same window
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        expectActiveWindow("kitty_A", '2');
+
+        // Dispatch-focus a different window
+        OK(getFromSocket("/dispatch focuswindow class:kitty_B"));
+        expectActiveWindow("kitty_B", '2');
+        OK(getFromSocket("/dispatch fullscreenstate 0 0"));
+
+        // Make a window that will request focus
+        if (!spawnKittyActivating())
+            return false;
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        OK(getFromSocket("/dispatch fullscreen 0"));
+        expectActiveWindow("kitty_A", '2');
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+        expectActiveWindow("kitty_activating", '2');
+    }
+
+    // Do nothing on conflict (but the dispatcher ignores this setting).
+    {
+        OK(getFromSocket("/keyword misc:on_focus_under_fullscreen 0"));
+
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        OK(getFromSocket("/dispatch fullscreen 0"));
+        expectActiveWindow("kitty_A", '2');
+
+        // Dispatch-focus the same window
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        expectActiveWindow("kitty_A", '2');
+
+        // Dispatch-focus a different window. When focused via the dispatcher,
+        // the value `misc:on_focus_under_fullscreen = 0` is treated like `= 1`
+        OK(getFromSocket("/dispatch focuswindow class:kitty_B"));
+        expectActiveWindow("kitty_B", '2');
+        OK(getFromSocket("/dispatch fullscreenstate 0 0"));
+
+        // Make a window that will request focus - the setting is treated normally
+        if (!spawnKittyActivating())
+            return false;
+        OK(getFromSocket("/dispatch focuswindow class:kitty_A"));
+        OK(getFromSocket("/dispatch fullscreen 0"));
+        expectActiveWindow("kitty_A", '2');
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
you absolutely cannot rely on timings in tests. Runners are _really_ slow and unpredictable. If you need to wait on _anything_ you need to poll something that will tell you exactly when an action finished.